backend/services/implementations/userService.ts

```diff
@@ -847,6 +842,354 @@ class UserService implements IUserService {
+          },
+        });
+
+        const { employee } = newUser;
+
+        return {
+          ...newUser,
+          id: String(newUser.id),
+          email: firebaseUser.email ?? "",
+          branchId: String(employee!.branchId),
+        };
+      } catch (postgresError) {
+        try {
+          await firebaseAdmin.auth().deleteUser(firebaseUser.uid);
+        } catch (firebaseError: unknown) {
+          const errorMessage = [
+            "Failed to rollback Firebase user creation after Postgres user creation failure. Reason =",
+            getErrorMessage(firebaseError),
+            "Orphaned authId (Firebase uid) =",
+            firebaseUser.uid,
+          ];
+          Logger.error(errorMessage.join(" "));
+        }
+
+        throw postgresError;
+      }
+    } catch (error: unknown) {
+      Logger.error(
+        `Failed to create EmployeeUser. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+  }
+
+  async updateEmployeeUserById(
+    userId: string,
+    employeeUser: UpdateEmployeeUserDTO,
+  ): Promise<EmployeeUserResponseDTO> {
+    let updatedFirebaseUser: firebaseAdmin.auth.UserRecord;
+
+    try {
+      const [oldEmployeeUser, updatedEmployeeUser] = await prisma.$transaction([
+        prisma.employee.findUnique({
+          where: {
+            id: Number(userId),
+          },
+          include: {
+            user: true,
+          },
+        }),
+        prisma.employee.update({
+          where: {
+            id: Number(userId),
+          },
+          data: {
+            user: {
+              update: {
+                firstName: employeeUser.firstName,
+                lastName: employeeUser.lastName,
+                role: "EMPLOYEE",
+                phoneNumber: employeeUser.phoneNumber,
+              },
+            },
+            branch: {
+              connect: { id: Number(employeeUser.branchId) },
+            },
+          },
+          include: {
+            user: true,
+          },
+        }),
+      ]);
+      try {
+        updatedFirebaseUser = await firebaseAdmin
+          .auth()
+          .updateUser(updatedEmployeeUser.user.authId, {
+            email: employeeUser.email,
+          });
+
+        const { user } = updatedEmployeeUser;
+
+        return {
+          ...user,
+          id: String(user.id),
+          email: updatedFirebaseUser.email ?? "",
+          branchId: String(updatedEmployeeUser.branchId),
+        };
+      } catch (error: unknown) {
+        try {
+          prisma.volunteer.update({
+            where: {
+              id: Number(userId),
+            },
+            data: {
+              user: {
+                update: {
+                  firstName: oldEmployeeUser!.user.firstName,
+                  lastName: oldEmployeeUser!.user.lastName,
+                  role: oldEmployeeUser!.user.role,
+                  phoneNumber: oldEmployeeUser!.user.phoneNumber,
+                },
+              },
```

Comment:
I think the rollback should also include reverting the branch ID right?

```suggestion
              },
              branch: {
                connect: { id: Number(oldEmployeeUser!.branchId) },
              },
```